zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java

```diff
@@ -1005,40 +1010,59 @@ private static List<String> applyUserSortField(
     return validationErrors;
   }
 
-  private static List<VariableValueFilter> toVariableValueFilters(
+  private static Either<List<String>, List<VariableValueFilter>> toVariableValueFiltersForUserTask(
       final List<UserTaskVariableFilterRequest> filters) {
+    if (CollectionUtils.isEmpty(filters)) {
+      return Either.right(null);
+    }
 
-    if (filters != null && !filters.isEmpty()) {
-      return filters.stream()
-          .map(
-              filter -> {
-                final var builder = new VariableValueFilter.Builder().name(filter.getName());
-                if (filter.getValue() != null) {
-                  builder.valueOperation(UntypedOperation.of(eq(filter.getValue())));
-                }
-                return builder.build();
-              })
-          .toList();
+    final List<GenericVariable<StringFilterProperty>> genericVariables = new ArrayList<>();
+    filters.forEach(
+        filter -> genericVariables.add(new GenericVariable<>(filter.getName(), filter.getValue())));
+    return toVariableValueFilters(genericVariables);
+  }
+
+  private static Either<List<String>, List<VariableValueFilter>>
+      toVariableValueFiltersForProcessInstance(
+          final List<ProcessInstanceVariableFilterRequest> filters) {
+    if (CollectionUtils.isEmpty(filters)) {
+      return Either.right(null);
     }
 
-    return null;
+    final List<GenericVariable<StringFilterProperty>> genericVariables = new ArrayList<>();
+    filters.forEach(
+        filter -> genericVariables.add(new GenericVariable<>(filter.getName(), filter.getValue())));
+    return toVariableValueFilters(genericVariables);
   }
 
-  private static List<VariableValueFilter> toVariableValueFiltersForProcessInstance(
-      final List<ProcessInstanceVariableFilterRequest> filters) {
-
-    if (filters != null && !filters.isEmpty()) {
-      return filters.stream()
-          .map(
-              filter ->
-                  new VariableValueFilter.Builder()
-                      .name(filter.getName())
-                      .valueOperation(UntypedOperation.of(eq(filter.getValue())))
-                      .build())
-          .toList();
+  private static Either<List<String>, List<VariableValueFilter>> toVariableValueFilters(
+      final List<GenericVariable<StringFilterProperty>> genericVariables) {
+    final List<String> validationErrors = new ArrayList<>();
+    final List<VariableValueFilter> variableValueFilters = new ArrayList<>();
+    genericVariables.forEach(
+        filter -> {
+          if (filter.name() == null) {
+            validationErrors.add(ERROR_MESSAGE_NULL_VARIABLE_NAME);
+          } else if (filter.value() == null) {
+            validationErrors.add(ERROR_MESSAGE_NULL_VARIABLE_VALUE);
+          } else if (validationErrors.isEmpty()) {
+            variableValueFilters.addAll(toVariableValueFilters(filter.name(), filter.value()));
+          }
```

Comment:
🔧 Both the `name` and the `value` can have validation errors. With the current if-else construct, we skip validating the `value` if the `name` has an issue.
```suggestion
          if (filter.name() == null) {
            validationErrors.add(ERROR_MESSAGE_NULL_VARIABLE_NAME);
          }
          if (filter.value() == null) {
            validationErrors.add(ERROR_MESSAGE_NULL_VARIABLE_VALUE);
          }
          if (validationErrors.isEmpty()) {
            variableValueFilters.addAll(toVariableValueFilters(filter.name(), filter.value()));
          }
```